src/thread_task.rs

```diff
@@ -710,19 +707,18 @@ fn delayed_fg_task<'l, 'ttd: 'l>(
         }
     }
     row = ttd.delayed_fg_progress[0].fetch_add(1, Ordering::SeqCst);
-    let _ = task_thread_lock.take();
+    drop(task_thread_lock.take());
     let delayed_fg = ttd.delayed_fg.try_read().unwrap();
     progmax = (delayed_fg.out.p.h + FG_BLOCK_SIZE as i32 - 1) / FG_BLOCK_SIZE as i32;
     loop {
         if (row + 1) < progmax {
             ttd.cond.notify_one();
         } else if row + 1 >= progmax {
-            *task_thread_lock = Some(ttd.lock.lock());
+            let _locked = ttd.lock.lock();
             ttd.delayed_fg_exec.set(0);
             if row >= progmax {
                 break;
             }
-            let _ = task_thread_lock.take();
```

Comment:
This doesn't look the same?  Is it?

Can you just do `drop(task_thread_lock.take())` at the end?

Reply:
Was this an automatic `cargo clippy --fix` change?

Reply:
I don't see the point of storing into `task_thread_lock` here?

Reply:
The former doesn't drop the lock `if row >= progmax`, but this does, right?

Reply:
Ahh, that's true.

Reply:
We have to be especially careful with the threading code here, because errors just make the program slower and won't cause any test failures.